src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -16511,23 +16512,37 @@ def dt_ceil(
         if nonexistent != "raise":
             ErrorMessage.parameter_not_implemented_error("nonexistent", method_name)
 
-        slice_length, slice_unit = rule_to_snowflake_width_and_slice_unit(
-            rule=freq  # type:  ignore[arg-type]
-        )
+        dtype = self.index_dtypes[0] if include_index else self.dtypes[0]
+        if is_datetime64_any_dtype(dtype):
+            slice_length, slice_unit = rule_to_snowflake_width_and_slice_unit(freq)
 
-        if slice_unit not in SUPPORTED_DT_FLOOR_CEIL_FREQS:
-            ErrorMessage.parameter_not_implemented_error(f"freq='{freq}'", method_name)
+            if slice_unit not in SUPPORTED_DT_FLOOR_CEIL_FREQS:
+                ErrorMessage.parameter_not_implemented_error(
+                    f"freq='{freq}'", method_name
+                )
+            return_type = None
 
-        def ceil_func(column: SnowparkColumn) -> SnowparkColumn:
-            floor_column = builtin("time_slice")(
-                column, slice_length, slice_unit, "START"
-            )
-            ceil_column = builtin("time_slice")(column, slice_length, slice_unit, "END")
-            return iff(column.equal_null(floor_column), column, ceil_column)
+            def ceil_func(column: SnowparkColumn) -> SnowparkColumn:
+                floor_column = builtin("time_slice")(
+                    column, slice_length, slice_unit, "START"
+                )
+                ceil_column = builtin("time_slice")(
+                    column, slice_length, slice_unit, "END"
+                )
+                return iff(column.equal_null(floor_column), column, ceil_column)
+
+        else:  # timedelta type
```

Comment:
Can we add an assert statement here verifying the index type as Timedelta?

Reply:
done